### ISSUE TITLE
Per-module deploy time configuration

### ISSUE THREAD

**Issue by ahal** [bnjbvr/trinity#26]
It may be desirable to configure some modules at deploy time, e.g to add credentials, support defaults or tweak the behaviour of the module. While it's currently possible to do this via the `!admin` module (see mastodon module for example), this isn't very convenient and the configuration is lost if the store is ever removed.

Instead, it would be nice if there existed some config file which could set arbitrary configuration per-module. Trinity would then pass the configuration for a specific module into that module on initial load. The modules would be responsible for taking care of the rest.

**Comment by bnjbvr** [comment]
Makes sense to me. If we implement this, it would be nice if it could be kept optional (i.e. doesn't crash the entire program if the configuration file is not there -- so it might be an optional command line argument or something like that). I like that the `!admin` module basically allows saving state at runtime, so I don't have to share secrets publicly in configuration files (speaking thereof... toml files, maybe?). If you're interested in implementing this, go ahead, I'm happy to review this later.
